### PR TITLE
Updated elixir bob mentoring notes

### PR DESCRIPTION
**Pull request by chriseyre2000**
Mostly formatting.

### CODE REVIEW

File: tracks/elixir/exercises/bob/mentoring.md [hunk 1]
```diff
@@ -32,6 +32,11 @@ end
 
 ### Common suggestions
 
+#### Hardcoding test data
+
+This is the first solution on the track and students will frequently hardcode test data to get a solution.
```
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
As first real exercise on the track, students will frequently hardcode test data to get to a solution.
```

File: tracks/elixir/exercises/bob/mentoring.md [hunk 2]
```diff
@@ -32,6 +32,11 @@ end
...
+Generally they should be recomended to try to find a more general solution.
```
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
They should be recommended to find a more general solution. 
```

File: tracks/elixir/exercises/bob/mentoring.md [hunk 3]
```diff
@@ -53,10 +58,12 @@ This exercise can be solved without regular expressions and a student could be e
 
 #### Hint functions for the truely stuck
 
-https://hexdocs.pm/elixir/String.html#trim/1
-https://hexdocs.pm/elixir/String.html#ends_with?/2
-https://hexdocs.pm/elixir/String.html#downcase/2
-https://hexdocs.pm/elixir/String.html#upcase/2
+- https://hexdocs.pm/elixir/String.html#trim/1
```
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- [String#trim/1](https://hexdocs.pm/elixir/String.html#trim/1)
```

File: tracks/elixir/exercises/bob/mentoring.md [hunk 4]
```diff
@@ -53,10 +58,12 @@ This exercise can be solved without regular expressions and a student could be e
...
+- https://hexdocs.pm/elixir/String.html#ends_with?/2
```
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- [String#ends_with?/2](https://hexdocs.pm/elixir/String.html#ends_with?/2)
```

File: tracks/elixir/exercises/bob/mentoring.md [hunk 5]
```diff
@@ -53,10 +58,12 @@ This exercise can be solved without regular expressions and a student could be e
...
+- https://hexdocs.pm/elixir/String.html#downcase/2
```
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- [String#downcase/2](https://hexdocs.pm/elixir/String.html#downcase/2)
```

File: tracks/elixir/exercises/bob/mentoring.md [hunk 6]
```diff
@@ -53,10 +58,12 @@ This exercise can be solved without regular expressions and a student could be e
...
+- https://hexdocs.pm/elixir/String.html#upcase/2
```
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- [String#upcase/2](https://hexdocs.pm/elixir/String.html#upcase/2)
```

File: tracks/elixir/exercises/bob/mentoring.md [hunk 7]
```diff
@@ -53,10 +58,12 @@ This exercise can be solved without regular expressions and a student could be e
...
+
+- String.upcase(input) != String.downcase(input) if there is a letter present.
```
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- `String.upcase(input) != String.downcase(input)` if there is a letter present.
```

File: tracks/elixir/exercises/bob/mentoring.md (hunk 3 above)
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- [`String#trim/1`](https://hexdocs.pm/elixir/String.html#trim/1)
```

File: tracks/elixir/exercises/bob/mentoring.md (hunk 4 above)
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- [`String#ends_with?/2`](https://hexdocs.pm/elixir/String.html#ends_with?/2)
```

File: tracks/elixir/exercises/bob/mentoring.md (hunk 5 above)
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- [`String#downcase/2`](https://hexdocs.pm/elixir/String.html#downcase/2)
```

File: tracks/elixir/exercises/bob/mentoring.md (hunk 6 above)
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- [`String#upcase/2`](https://hexdocs.pm/elixir/String.html#upcase/2)
```

File: tracks/elixir/exercises/bob/mentoring.md (hunk 7 above)
Comment (SleeplessByte) [inline tracks/elixir/exercises/bob/mentoring.md]:
```suggestion
- `String.upcase(input) != String.downcase(input)` if there is a letter present.
```

**Comment by rpottsoh** [comment]
Pinging @chriseyre2000

**Pull request merged**